### PR TITLE
Substation Clustering 

### PR DESCRIPTION
**Pull request by BamunugeDR99**
Substation Clustering using Louvain Algorithm

### CODE REVIEW

File: grid2op/multi_agent/utils.py [hunk 1]
```diff
@@ -74,4 +78,77 @@ def __eq__(self, other):
             and self._current_agent == other._current_agent
             and self.selected_agent == other.selected_agent
         )
-        
\ No newline at end of file
+
+
+class ClusterUtils:
+    """
+    Utility class for clustering substations
```
Comment (Gajithra) [inline grid2op/multi_agent/utils.py]: Suggestion: """ Outputs clustered substation based on the Louvain graph clustering method"""

Just to give an initial overview of the functionality of this utility class.

File: grid2op/multi_agent/utils.py [hunk 2]
```diff
@@ -74,4 +78,77 @@ def __eq__(self, other):
...
+
+        The connectivity matrix is a 2D NumPy array where the element at position (i, j) is 1 if there is a direct 
+        connection between substation i and substation j, and 0 otherwise. The diagonal elements are set to 1 to indicate 
+        self-connections.
+
+        Args:
+            env (grid2op.Environment): The grid environment for which the connectivity matrix is to be created.
+
+        Returns:
+            np.ndarray(env.n_sub, env.n_sub): A 2D NumPy array representing the connectivity matrix of the grid environment.
```
Comment (Gajithra) [inline grid2op/multi_agent/utils.py]: Returns:
      - connectivity_matrix: A 2D Numpy array of dimension (env.n_sub, env.n_sub) representing the substation connectivity of the grid environment.

File: grid2op/multi_agent/utils.py (hunk 1 above)
Comment (BamunugeDR99) [reply grid2op/multi_agent/utils.py]: Updated the docstring

File: grid2op/multi_agent/utils.py (hunk 2 above)
Comment (BamunugeDR99) [reply grid2op/multi_agent/utils.py]: Updated the docstring

**Review by Gajithra — APPROVED** [review]
Looks good. Left just a couple of suggestions for your reference

**Review by Thirunayan22 — APPROVED** [review]

**Pull request merged**